chunky/src/java/se/llbit/chunky/resources/Texture.java

```diff
@@ -1122,6 +1122,100 @@ public class Texture {
+  @TexturePath("assets/minecraft/textures/block/cherry_door_bottom")
+  public static final Texture cherryDoorBottom = new Texture();
+  @TexturePath("assets/minecraft/textures/block/cherry_leaves")
+  public static final Texture cherryLeaves = new Texture();
+  @TexturePath("assets/minecraft/textures/block/cherry_log")
+  public static final Texture cherryLog = new Texture();
+  @TexturePath("assets/minecraft/textures/block/cherry_log_top")
+  public static final Texture cherryLogTop = new Texture();
+  @TexturePath("assets/minecraft/textures/block/stripped_cherry_log")
+  public static final Texture strippedCherryLog = new Texture();
+  @TexturePath("assets/minecraft/textures/block/stripped_cherry_log_top")
+  public static final Texture strippedCherryLogTop = new Texture();
+  @TexturePath("assets/minecraft/textures/entity/signs/cherry")
+  public static final Texture cherrySignPost = new Texture();
+  @TexturePath("assets/minecraft/textures/block/cherry_trapdoor")
+  public static final Texture cherryTrapdoor = new Texture();
+  @TexturePath("assets/minecraft/textures/block/cherry_sapling")
+  public static final Texture cherrySapling = new Texture();
+  @TexturePath("assets/minecraft/textures/block/torchflower")
+  public static final Texture torchflower = new Texture();
+  @TexturePath("assets/minecraft/textures/block/torchflower_crop_stage0")
+  public static final Texture torchflowerCropStage0 = new Texture();
+  @TexturePath("assets/minecraft/textures/block/torchflower_crop_stage1")
+  public static final Texture torchflowerCropStage1 = new Texture();
+  @TexturePath("assets/minecraft/textures/block/torchflower_crop_stage2")
+  public static final Texture torchflowerCropStage2 = new Texture();
+  @TexturePath("assets/minecraft/textures/block/chiseled_bookshelf_empty")
+  public static final Texture chiseledBookshelfEmpty = new Texture();
+  @TexturePath("assets/minecraft/textures/block/chiseled_bookshelf_occupied")
+  public static final Texture chiseledBookshelfOccupied = new Texture();
+  @TexturePath("assets/minecraft/textures/block/chiseled_bookshelf_side")
+  public static final Texture chiseledBookshelfSide = new Texture();
+  @TexturePath("assets/minecraft/textures/block/chiseled_bookshelf_top")
+  public static final Texture chiseledBookshelfTop = new Texture();
+
+  public static final Texture[] chiseledBookshelfCombinations = new OverlaidTexture[64];
+  static {
+    for(int i = 0; i < chiseledBookshelfCombinations.length; i++) {
+      int j = i; // it complains if I don't do this
+      chiseledBookshelfCombinations[i] = new OverlaidTexture(Texture.chiseledBookshelfEmpty, Texture.chiseledBookshelfOccupied, (x, y) ->
+        ((j & 1) != 0 && x < 1f/3 && y < 1f/2) || ((j & 2) != 0 && x >= 1f/3 && x < 2f/3 && y < 1f/2) || ((j & 4) != 0 && x >= 2f/3 && y < 1f/2)
+        || ((j & 8) != 0 && x < 1f/3 && y >= 1f/2) || ((j & 16) != 0 && x >= 1f/3 && x < 2f/3 && y >= 1f/2) || ((j & 32) != 0 && x >= 2f/3 && y >= 1f/2)
+      );
```

Comment:
Can you refactor this so that it is readable?

Reply:
Done, let me know if it is good